app/jobs/regular/shared_conversation_adjust_upload_security.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ::Jobs
+  class SharedConversationAdjustUploadSecurity < ::Jobs::Base
+    def execute(args)
+      # The conversation context includes post cooked content so this
+      # must be updated when target uploads security changes.
+      update_conversation(args[:conversation_id]) if args[:conversation_id].present?
+
+      # If we deleted the conversation then we just need to update the target's
+      # uploads security, no need to update the conversation.
+      if args[:target_id].present? && args[:target_type].present?
+        update_target(args[:target_id], args[:target_type])
+      end
```

Comment:
I assume the call to `update_conversation` and `update_target` are intended to be mutually exclusive? If so, can we do an `if elsif` to make the case of both of them running impossible?

Reply:
Yes that's right, and sure I can change this :+1: